metamorph/processor_response/processor_response.go

```diff
@@ -19,126 +17,68 @@ type StatusAndError struct {
 	mu     deadlock.RWMutex
 	Err    error                `json:"err"`
 	Status metamorph_api.Status `json:"status"`
 }
 
 func NewProcessorResponse(hash *chainhash.Hash) *ProcessorResponse {
-	return newProcessorResponse(hash, metamorph_api.Status_RECEIVED, nil)
+	return newProcessorResponse(hash, metamorph_api.Status_RECEIVED, nil, time.Second)
 }
 
-// NewProcessorResponseWithStatus creates a new ProcessorResponse with the given status.
-// It is used when restoring the ProcessorResponseMap from the database.
-func NewProcessorResponseWithStatus(hash *chainhash.Hash, status metamorph_api.Status) *ProcessorResponse {
-	return newProcessorResponse(hash, status, nil)
+func NewProcessorResponseWithChannel(hash *chainhash.Hash, ch chan StatusAndError, timeout time.Duration) *ProcessorResponse {
+	return newProcessorResponse(hash, metamorph_api.Status_RECEIVED, ch, timeout)
 }
 
-func NewProcessorResponseWithChannel(hash *chainhash.Hash, ch chan StatusAndError) *ProcessorResponse {
-	return newProcessorResponse(hash, metamorph_api.Status_RECEIVED, ch)
-}
-
-func newProcessorResponse(hash *chainhash.Hash, status metamorph_api.Status, ch chan StatusAndError) *ProcessorResponse {
+func newProcessorResponse(hash *chainhash.Hash, status metamorph_api.Status, ch chan StatusAndError, timeout time.Duration) *ProcessorResponse {
 	pr := &ProcessorResponse{
-		Start:          time.Now(),
-		Hash:           hash,
-		Status:         status,
-		callerCh:       ch,
-		statusUpdateCh: make(chan *ProcessorResponseStatusUpdate, 10),
+		Hash:     hash,
+		Status:   status,
+		callerCh: ch,
 	}
-	pr.LastStatusUpdateNanos.Store(pr.Start.UnixNano())
-
-	go func() {
-		for statusUpdate := range pr.statusUpdateCh {
-			pr.updateStatus(statusUpdate)
-		}
-	}()
 
 	return pr
 }
 
-func (r *ProcessorResponse) Close() {
-	defer func() {
-		_ = recover()
-	}()
-
-	if r.statusUpdateCh != nil {
-		close(r.statusUpdateCh)
-	}
-}
-
 func (r *ProcessorResponse) UpdateStatus(statusUpdate *ProcessorResponseStatusUpdate) {
-	r.statusUpdateCh <- statusUpdate
-}
-
-func (r *ProcessorResponse) updateStatus(statusUpdate *ProcessorResponseStatusUpdate) {
-	// If this transaction has already been mined, ignore any further updates
-	if r.Status == metamorph_api.Status_MINED {
-		return
-	}
-
-	if statusUpdate.UpdateStore != nil {
-		if err := statusUpdate.UpdateStore(); err != nil {
-			r.setErr(err, "processorResponse")
-			statusUpdate.Callback(err)
-			return
-		}
-	}
-
 	if statusUpdate.StatusErr != nil {
 		_ = r.setStatusAndError(statusUpdate.Status, statusUpdate.StatusErr)
 	} else {
 		_ = r.setStatus(statusUpdate.Status)
 	}
 
-	statKey := fmt.Sprintf("%d: %s", statusUpdate.Status, statusUpdate.Status.String())
-	r.LastStatusUpdateNanos.Store(gocore.NewStat("processorResponse").NewStat(statKey).AddTime(r.LastStatusUpdateNanos.Load()))
-
-	if !statusUpdate.IgnoreCallback {
-		statusUpdate.Callback(nil)
-	}
 }
 
 func (r *ProcessorResponse) String() string {
```

Comment:
I think we can remove this function as it is only used here: `metamorph/processor_response/processor_response_test.go`

```

func TestString(t *testing.T) {
	t.Run("String", func(t *testing.T) {
		response := NewProcessorResponse(testdata.TX1Hash)
		assert.IsType(t, "string", response.String())
	})
}
```